src/plugin/withSentryIOS.ts

```diff
@@ -0,0 +1,70 @@
+import { ConfigPlugin, withDangerousMod, withXcodeProject } from '@expo/config-plugins';
+import * as fs from 'fs';
+import * as path from 'path';
+
+export const withSentryIOS: ConfigPlugin<string> = (config, sentryProperties: string) => {
+  config = withXcodeProject(config, (config) => {
+    const xcodeProject = config.modResults;
+
+    const sentryBuildPhase = xcodeProject.pbxItemByComment(
+      'Upload Debug Symbols to Sentry',
+      'PBXShellScriptBuildPhase'
+    );
+    if (!sentryBuildPhase) {
+      xcodeProject.addBuildPhase(
+        [],
+        'PBXShellScriptBuildPhase',
+        'Upload Debug Symbols to Sentry',
+        null,
+        {
+          shellPath: '/bin/sh',
+          shellScript:
+            'export SENTRY_PROPERTIES=sentry.properties\\n' +
+            '../node_modules/@sentry/cli/bin/sentry-cli upload-dsym',
+        }
+      );
+    }
+
+    let bundleReactNativePhase = xcodeProject.pbxItemByComment(
+      'Bundle React Native code and images',
+      'PBXShellScriptBuildPhase'
+    );
+    modifyExistingXcodeBuildScript(bundleReactNativePhase);
+
+    return config;
+  });
+
+  return withDangerousMod(config, [
+    'ios',
+    (config) => {
+      writeSentryPropertiesTo(path.resolve(config.modRequest.projectRoot, 'ios'), sentryProperties);
+      return config;
+    },
+  ]);
+};
+
+function modifyExistingXcodeBuildScript(script: any): void {
+  if (
+    !script.shellScript.match(/(packager|scripts)\/react-native-xcode\.sh\b/) ||
+    script.shellScript.match(/sentry-cli\s+react-native[\s-]xcode/)
+  ) {
+    return;
+  }
+  let code = JSON.parse(script.shellScript);
+  code =
+    'export SENTRY_PROPERTIES=sentry.properties\n' +
+    'export EXTRA_PACKAGER_ARGS="--sourcemap-output $DERIVED_FILE_DIR/main.jsbundle.map"\n' +
+    code.replace(
+      /^.*?\/(packager|scripts)\/react-native-xcode\.sh\s*/m,
+      (match: any) => `../node_modules/@sentry/cli/bin/sentry-cli react-native xcode ${match}`
+    );
+  script.shellScript = JSON.stringify(code);
+}
+
+export function writeSentryPropertiesTo(filepath: string, sentryProperties: string) {
+  if (!fs.existsSync(filepath)) {
+    throw new Error("Directory '" + filepath + "' does not exist.");
+  }
+
+  fs.writeFileSync(path.resolve(filepath, 'sentry.properties'), sentryProperties);
```

Comment:
Does this file need to be added to the xcode project?

Reply:
It's just used in one of the build-phase shell scripts to grab properties, so it shouldn't need to be added to Xcode 

sentry's own scripts for adding this via their "wizard" tool (which is a sick name btw) don't add this to the xcode project files